akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/RouteDirectivesSpec.scala

```diff
@@ -133,16 +137,17 @@ class RouteDirectivesSpec extends FreeSpec with GenericRoutingSpec {
 
   case class Data(name: String, age: Int)
   object Data {
-    //import spray.json.DefaultJsonProtocol._
-    //import spray.httpx.SprayJsonSupport._
+    import spray.json.DefaultJsonProtocol._
+    import SprayJsonSupport._
+    import ScalaXmlSupport._
 
-    val jsonMarshaller: ToEntityMarshaller[Data] = FIXME // jsonFormat2(Data.apply)
-    val xmlMarshaller: ToEntityMarshaller[Data] = FIXME
-    /*Marshaller.delegate[Data, xml.NodeSeq](MediaTypes.`text/xml`) { (data: Data) ⇒
-        <data><name>{ data.name }</name><age>{ data.age }</age></data>
-      }*/
+    val jsonMarshaller: ToEntityMarshaller[Data] = jsonFormat2(Data.apply)
+
+    val xmlMarshaller: ToEntityMarshaller[Data] = Marshaller.combined { (data: Data) ⇒
```

Comment:
Yes, nice, if you want the lookup to happen automatically that's indeed nicer. On the other hand, I also like the more explicit version which helps with reading the code.